src/components/offline/OfflineService.js

```diff
@@ -97,9 +63,54 @@
     initDownloadStatus();
 
 
-    this.$get = function($rootScope, $timeout, $translate, $window,
-        gaBrowserSniffer, gaGlobalOptions, gaLayers, gaMapUtils,
+    this.$get = function($http, $rootScope, $timeout, $translate, $window,
+        gaBrowserSniffer, gaGlobalOptions, gaKml, gaLayers, gaMapUtils,
```

Comment:
Hmm. I'm torn. gaKml is only needed to check if layer is kml, right (gaKml.isKmlLayer)? Maybe just create internal function that does the check instead of injecting a dependency?

As said, not sure here.